nifi-nar-bundles/nifi-site-to-site-reporting-bundle/nifi-site-to-site-reporting-task/src/main/java/org/apache/nifi/reporting/AbstractSiteToSiteReportingTask.java

```diff
@@ -119,8 +120,15 @@
     }
 
     @OnScheduled
-    public void setup(final ConfigurationContext context) throws IOException {
-        siteToSiteClient = SiteToSiteUtils.getClient(context, getLogger());
+    public void setup(final ConfigurationContext configContext) throws IOException {
+        this.configContext = configContext;
+    }
+
+    public void setup(final ReportingContext reportContext) throws IOException {
+        //Test doesn't run OnScheduled setup and provides a Mock SiteToSiteClient, so ignore it
+        if (configContext != null) {
```

Comment:
Related to my comment above, you don't have to check for `configContext != null`, instead you could check for `siteToSiteClient != null` as long as the ReportingTask sets the `siteToSiteClient` to null when it is disabled/stopped/shutdown.

Reply:
I've done with changes you proposed. Thank you!